jailman.sh

```diff
@@ -15,131 +111,120 @@ source "${SCRIPT_DIR}/includes/global.sh"
-            echo "   -u to update jails, listed by name, space seperated like this: jackett plex sonarr"
-            echo "   -d to destroy jails, listed by name, space seperated like this: jackett plex sonarrt"
-            echo "   -g to upgrade jails, listed by name, space seperated like this: jackett plex sonarr"
-            echo "   -h help (this output)"
-            exit 0
-            ;;
-	? ) echo "Error: Invalid option was specified -$OPTARG"
-            exit 0
-            ;;
-     esac
+do
+	#Shellcheck on wordsplitting will be disabled. Wordsplitting can't happen, because it's already split using OPTIND.
+	case $opt in
+		i ) installjails=("$OPTARG")
+			until (( OPTIND > arglen )) || [[ ${args[$OPTIND-1]} =~ ^-.* ]]; do
+				installjails+=("${args[$OPTIND-1]}")
+				OPTIND=$((OPTIND + 1))
+			done
+			;;
+		r ) redojails=("$OPTARG")
+			until (( OPTIND > arglen )) || [[ ${args[$OPTIND-1]} =~ ^-.* ]]; do
+				redojails+=("${args[$OPTIND-1]}")
+				OPTIND=$((OPTIND + 1))
+			done
+			;;
+		u ) updatejails=("$OPTARG")
+			until (( OPTIND > arglen )) || [[ ${args[$OPTIND-1]} =~ ^-.* ]]; do
+				updatejails+=("${args[$OPTIND-1]}")
+				OPTIND=$((OPTIND + 1))
+			done
+			;;
+		d ) destroyjails=("$OPTARG")
+			until (( OPTIND > arglen )) || [[ ${args[$OPTIND-1]} =~ ^-.* ]]; do
+				destroyjails+=("${args[$OPTIND-1]}")
+				OPTIND=$((OPTIND + 1))
+			done
+			;;
+		g ) upgradejails=("$OPTARG")
+			until (( OPTIND > arglen )) || [[ ${args[$OPTIND-1]} =~ ^-.* ]]; do
+				upgradejails+=("${args[$OPTIND-1]}")
+				OPTIND=$((OPTIND + 1))
+			done
+			;;
+		h ) 
+			usage
+			exit 3
+			;;
+		* ) echo "Error: Invalid option was specified -$OPTARG"
+			usage
+			exit 3
+			;;
+	esac
 done
 
 # auto detect iocage install location
-global_dataset_iocage=$(zfs get -H -o value mountpoint $(iocage get -p)/iocage)
+global_dataset_iocage=$(zfs get -H -o value mountpoint "$(iocage get -p)"/iocage)
 global_dataset_iocage=${global_dataset_iocage#/mnt/}
 export global_dataset_iocage
 
 # Parse the Config YAML
+for configpath in "${SCRIPT_DIR}"/blueprints/*/config.yml; do
+	# shellcheck disable=SC2046
+	! eval $(parse_yaml "${configpath}")
+done
+
 # shellcheck disable=SC2046
-for configpath in "${SCRIPT_DIR}"/blueprints/*/config.yml; do ! eval $(parse_yaml "${configpath}"); done
 eval $(parse_yaml "${SCRIPT_DIR}/includes/global.yml")
+# shellcheck disable=SC2046
 eval $(parse_yaml "${SCRIPT_DIR}/config.yml")
 
-if [ "${global_version}" != "1.3" ]; then
+if [ "${global_version:-}" != "1.3" ]; then
 	echo "You are using old config.yml synatx."
 	echo "Please check the wiki for required changes"
 	exit 1
 fi
 
 # Check and Execute requested jail destructions
-if [ ${#destroyjails[@]} -eq 0 ]; then 
-	echo "No jails to destroy"
-else
+if [ ${#destroyjails[@]} -gt 0 ]; then
 	# shellcheck disable=SC2124,SC2145
 	echo "jails to destroy ${destroyjails[@]}"
```

Comment:
This should be fixable with:
```suggestion
	echo "jails to destroy " "${destroyjails[@]}"
```

According to:
https://github.com/koalaman/shellcheck/wiki/SC2145

Reply:
missed these, looking...